simpleclient/src/main/java/io/prometheus/client/Summary.java

```diff
@@ -38,133 +40,231 @@
  * This would allow you to track request rate, average latency and average request size.
  */
 public class Summary extends SimpleCollector<Summary.Child> {
+    // Quantiles that should be calculated for this Summary
+    // e.g.  0.20, 0.50, 0.99
+    private Quantile[] quantiles;
+    private Stream stream;
+    private boolean shouldCalculateQuant = false;
 
-  Summary(Builder b) {
-    super(b);
-  }
+    Summary(Builder b) {
+        super(b);
+        quantiles = b.quantiles;
+        this.stream = b.stream;
+        this.shouldCalculateQuant = b.shouldCalculateQuant;
+    }
 
-  public static class Builder extends SimpleCollector.Builder<Builder, Summary> {
-    @Override
-    public Summary create() {
-      return new Summary(this);
+    public static class Builder extends SimpleCollector.Builder<Builder, Summary> {
+        private Quantile[] quantiles = new Quantile[]{new Quantile(0.2, 0.05), new Quantile(0.5, 0.1),
+                new Quantile(0.9, 0.01)};
+
+        private Stream<Double> stream;
+        private boolean shouldCalculateQuant;
+        @Override
+        public Summary create() {
+            return new Summary(this);
+        }
+
+        public Builder quantiles(Quantile... quantiles) {
+            // We want the Summary to display quantiles in ascending order (e.g. 0.20, 0.50, 0.99)
+            //Arrays.sort(quantiles);
+            this.quantiles = quantiles;
+            this.shouldCalculateQuant = true;
+            return this;
+        }
+
+        public Builder stream(Stream stream) {
```

Comment:
I've seen the equivalent of stream being called *reservoir* in several places, is that a more accurate term?